squire_lib/src/accounts.rs

```diff
@@ -0,0 +1,29 @@
+use uuid::Uuid;
+use mtgjson::model::deck::Deck;
+use settings;
+
+enum SharingPermissions {
```

Comment:
Ok, so we want to allow for players to specify how much information they are ok with sharing after a tournament has concluded. We want to give them the ability to toggle how much deck info they are sharing. Per your comment, we would be checking the fields of `decklist` to determine how much info is being shared. This works, but there is a better way to go about this.

In Rust, we can have an enum like this:
```rust
enum SomeEnum {
  VariantA,
  VariantB,
  VariantC,
}
```
This enum is essentially a three-value type. That is, if I have some variable whose type is `SomeEnum`, then that variable can only be one of exactly three values (`VariantA` and so on). I think that is all we need here.

I don't think this enum needs to hold any data since we probably don't want to use it to swap data around or anything. Instead, we are just trying to encode "how much information do I want to share". The following would probably suffice:
```suggestion
pub enum SharingPermissions {
  Everything,
  OnlyDeckInfo,
  OnlyDeckName,
  Nothing
}
```